File: python_proj/mylib/query.py
"""Query the database"""

import sqlite3


def query():
    """Query the database and perform insert, update, and delete operations"""
    conn = sqlite3.connect("../data/US_births_DB.db")
    cursor = conn.cursor()

    # Perform SELECT
    cursor.execute("SELECT * FROM US_births_DB")

    # Perform INSERT
    cursor.execute(
        """
        INSERT INTO US_births_DB (year, month, date_of_month, day_of_week, births) 
        VALUES (2008, 8, 8, 1, 9999)"""
    )

    # Perform UPDATE
    cursor.execute(
        """
        UPDATE US_births_DB 
        SET day_of_week = 1, births = 6666
        WHERE id = 1"""
    )

    # Perform DELETE
    cursor.execute(
        """
        DELETE FROM US_births_DB 
        WHERE id = 2"""
    )

    conn.commit()  # Ensure all changes are saved
    conn.close()
    return "Success"  # Return success for testing

File: rust_proj/src/query.rs
use rusqlite::{Connection, Result};

pub fn query() -> Result<String> {
    // Connect to database
    let conn = Connection::open("../data/US_births_DB.db")?;

    // Execute SELECT
    let mut stmt = conn.prepare("SELECT * FROM US_births_DB")?;
    let _rows = stmt.query([])?;
    // If you need to process the results, you can iterate over rows

    // Execute INSERT
    conn.execute(
        "INSERT INTO US_births_DB (year, month, date_of_month, day_of_week, births) 
         VALUES (?1, ?2, ?3, ?4, ?5)",
        [2008, 8, 8, 1, 9999],
    )?;

    // Execute UPDATE
    conn.execute(
        "UPDATE US_births_DB 
         SET day_of_week = ?1, births = ?2
         WHERE id = ?3",
        [1, 6666, 1],
    )?;

    // Execute DELETE
    conn.execute(
        "DELETE FROM US_births_DB 
         WHERE id = ?1",
        [2],
    )?;

    // Commit changes will be automatically executed when the Connection is dropped
    Ok("Success".to_string())
}
